### PR TITLE
MGMT-10127: Enhance logging for package utils

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**

### CODE REVIEW

File: pkg/utils/hash.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ func Annotate(obj *unstructured.Unstructured) error {
 
 	hash, err := hashstructure.Hash(obj.Object, hashstructure.FormatV2, nil)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to hash object %s/%s: %w", obj.GetNamespace(), obj.GetName(), err)
```
Comment (ybettan) [inline pkg/utils/hash.go:25]:
```suggestion
		return fmt.Errorf("failed to hash %s/%s in namespace %s: %w", obj.GetKind(), obj.GetName(), obj.GetNamespace(), err)
```

File: pkg/utils/hash.go [hunk 2]
```diff
@@ -37,7 +37,7 @@ func AnnotationEqual(new *unstructured.Unstructured, old *unstructured.Unstructu
 
 	hash, err := hashstructure.Hash(old.Object, hashstructure.FormatV2, nil)
 	if err != nil {
-		return false, err
+		return false, fmt.Errorf("failed to hash old object %s/%s: %w", old.GetNamespace(), old.GetName(), err)
```
Comment (ybettan) [inline pkg/utils/hash.go:40]:
```suggestion
		return false, fmt.Errorf("failed to hash old %s/%s in namespace %s: %w", old.GetKind(), old.GetName(), old.GetNamespace(), err)
```

File: pkg/utils/osversion.go [hunk 3]
```diff
@@ -17,7 +15,7 @@ func ParseOSInfo(osImagePretty string) (string, string, string, error) {
 	clusterVersion, osVersion, osMajor := "", "", ""
 	matches := versionRegex.FindStringSubmatch(osImagePretty)
 	if len(matches) != 3 {
-		return clusterVersion, osVersion, osMajor, errors.New("unable to find a match " + osImagePretty)
+		return clusterVersion, osVersion, osMajor, fmt.Errorf("failed to find a match for %s", osImagePretty)
```
Comment (ybettan) [inline pkg/utils/osversion.go:18]:
```suggestion
		return "", "", "", fmt.Errorf("failed to find a match for %s", osImagePretty)
```

Just for clearness.

File: pkg/utils/hash.go (hunk 1 above)
Comment (ybettan) [reply pkg/utils/hash.go:25]: According to your other PRs, you will probably prefer to keep it.

File: pkg/utils/hash.go (hunk 2 above)
Comment (ybettan) [reply pkg/utils/hash.go:40]: According to your other PRs, you will probably prefer to keep it. Let us at least replace `object` with `obj.GetKind()` ?

File: pkg/utils/hash.go (hunk 1 above)
Comment (ybettan) [reply pkg/utils/hash.go:25]:
```suggestion
		return fmt.Errorf("failed to hash %s %s/%s: %w", obj.GetKind(), obj.GetNamespace(), obj.GetName(), err)
```

File: pkg/utils/hash.go (hunk 2 above)
Comment (ybettan) [reply pkg/utils/hash.go:40]:
```suggestion
		return false, fmt.Errorf("failed to hash old %s %s/%s: %w", old.GetKind(), old.GetNamespace(), old.GetName(), err)
```

File: pkg/utils/osversion.go (hunk 3 above)
Comment (yevgeny-shnaidman) [reply pkg/utils/osversion.go:18]: fixed

File: pkg/utils/hash.go (hunk 2 above)
Comment (yevgeny-shnaidman) [reply pkg/utils/hash.go:40]: fixed

**Pull request merged**